### ISSUE TITLE
Calling `super` inside a module function compiles to js that references an undefined variable

### ISSUE THREAD

**Issue by vais** [opal/opal#935]
The following Ruby:
```ruby
module M
  def self.meth
    super
  end
end
```
Compiles to the following JS:
```javascript
(function(Opal) {
  Opal.dynamic_require_severity = "error";
  var self = Opal.top, $scope = Opal, nil = Opal.nil, $breaker = Opal.breaker, $slice = Opal.slice, $module = Opal.module;

  return (function($base) {
    var self = $module($base, 'M');

    var def = self.$$proto, $scope = self.$$scope, TMP_1;

    Opal.defs(self, '$meth', TMP_1 = function() {var $zuper = $slice.call(arguments, 0);
      var self = this, $iter = TMP_1.$$p, $yield = $iter || nil;

      TMP_1.$$p = null;
      return Opal.find_super_dispatcher(self, 'meth', TMP_1, $iter, $M).apply(self, $zuper);
    })
  })(self)
})(Opal);
```
Note the reference to the undefined variable `$M` in the `Opal.find_super_dispatcher` call.

**Comment by elia** [comment]
This should be reverted if possible 

https://github.com/vais/opal/commit/4c619d48dbe1c73e4ec5a576fc33c84496cce2f4

**Comment by elia** [comment]
Fixed by #1217
